test/integration_test.py

```diff
@@ -641,6 +630,60 @@ def worker(_):
                     expect_execution = fetch == first_fetch
                     self.assertEqual(1 if expect_execution else 0, len(execution_ids))
 
+    def _manifest_filters(self, catalog: CatalogName) -> JSON:
+        # IT catalogs with just one public source are always indexed completely
+        # if that source contains less than the minimum number of bundles
+        # required. So regardless of any randomness employed by this test,
+        # manifests derived from these catalogs will always be based on the same
+        # content hash. Since the resulting reuse of cached manifests interferes
+        # with this test, we need another means of randomizing the manifest key:
+        # a random but all-inclusive filter.
+        tibi_byte = 1024 ** 4
+        return {
+            self._file_size_facet(catalog): {
+                'within': [[0, tibi_byte + self.random.randint(0, tibi_byte)]]
+            }
+        }
+
+    def _test_manifest_tagging_race(self, catalog: CatalogName):
+        validators: dict[ManifestFormat, Callable[[CatalogName, bytes], None]] = {
+            ManifestFormat.compact: self._check_compact_manifest,
+            ManifestFormat.curl: self._check_curl_manifest,
+        }
+
+        for format, validator in validators.items():
+            with self.subTest('manifest_tagging_race', catalog=catalog, format=format):
+                filters = self._manifest_filters(catalog)
+                manifest_url = config.service_endpoint.set(path='/manifest/files',
+                                                           args=dict(catalog=catalog,
+                                                                     filters=json.dumps(filters),
+                                                                     format=format.value))
+                method = PUT
+                responses = []
+                while True:
+                    response = self._get_url(method, manifest_url)
+                    if response.status == 301:
+                        responses.append(response)
+                        # Request the same manifest without following the
+                        # redirect in order to expose the race condition that
+                        # returns an untagged cached manifest. This happens
+                        # when a step function execution has generated a
+                        # manifest but is still in the process of tagging the
+                        # object. Sleeping .33s for each request in order to
+                        # prevent tripping the rate limit. With more requests
+                        # the likelihood of catching a step function in the
+                        # racy state is almost guraranteed.
+                        time.sleep(.33)
```

Comment:
```suggestion
                        time.sleep(.33)
```

Brittle. Derive from the actual rate limit.